plugins/tiledb/io/TileDBReader.cpp

```diff
@@ -312,6 +318,10 @@ void TileDBReader::localReady()
         if (numDims == 2)
             m_query->set_subarray({m_bbox.minx, m_bbox.maxx,
                 m_bbox.miny, m_bbox.maxy});
+        else if (numDims == 4 && m_has_time)
```

Comment:
This is a little weird. If you have 4 dims and no time, do you want it to fall into the case below? Should there be an explicit test for the case below rather than `else`?